File: product/src/main/java/com/sparta/product/infrastructure/config/EcsConfig.java
package com.sparta.product.infrastructure.config;

import lombok.extern.slf4j.Slf4j;
import org.springframework.cloud.commons.util.InetUtils;
import org.springframework.cloud.netflix.eureka.EurekaInstanceConfigBean;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.InetAddress;
import java.net.URL;

@Slf4j
@Configuration
public class EcsConfig {

	@Bean
	public EurekaInstanceConfigBean eurekaInstanceConfig(InetUtils inetUtils) {
		EurekaInstanceConfigBean config = new EurekaInstanceConfigBean(inetUtils);
		String publicIp = null;
		String privateIp;
		try {
			URL url = new URL("https://api.ipify.org");

			privateIp = InetAddress.getLocalHost().getHostAddress();
			BufferedReader reader = new BufferedReader(new InputStreamReader(url.openStream()));
			publicIp = reader.readLine();
			log.info("Public IP Address: {}", publicIp);
			log.info("Private IP Address: {}", privateIp);
		} catch (Exception e) {
			log.error("Failed : {}", e.getMessage());
		}

		config.setIpAddress(publicIp);

		return config;
	}

}

File: history/src/main/kotlin/com/study/history/infrastructure/config/EcsConfig.kt
package com.study.history.infrastructure.config

import org.springframework.cloud.commons.util.InetUtils
import org.springframework.cloud.netflix.eureka.EurekaInstanceConfigBean
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.io.BufferedReader
import java.io.InputStreamReader
import java.net.URL

@Configuration
class EcsConfig {

    companion object {
        private const val API_URL = "https://api.ipify.org"
        private val logger = LoggerProvider.logger
    }

    @Bean
    fun eurekaInstanceConfig(inetUtils: InetUtils): EurekaInstanceConfigBean {
        val eurekaInstanceConfigBean = EurekaInstanceConfigBean(inetUtils)

        val publicIp = try {
            val url = URL(API_URL)
            BufferedReader(InputStreamReader(url.openStream())).readLine()
        } catch (e: Exception) {
            logger.error { "Failed : ${e.message}" }
            ""
        }

        eurekaInstanceConfigBean.ipAddress = publicIp

        return eurekaInstanceConfigBean
    }

}